src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1108,13 +1108,15 @@ public static void applyPartitioner()
         {
             throw new ConfigurationException("Missing directive: partitioner", false);
         }
+        String partitionerName = conf.partitioner;
         try
         {
-            partitioner = FBUtilities.newPartitioner(System.getProperty(Config.PROPERTY_PREFIX + "partitioner", conf.partitioner));
+            partitionerName = System.getProperty(Config.PROPERTY_PREFIX + "partitioner", conf.partitioner);
+            partitioner = FBUtilities.newPartitioner(partitionerName);
         }
         catch (Exception e)
         {
-            throw new ConfigurationException("Invalid partitioner class " + conf.partitioner, false);
+            throw new ConfigurationException("Invalid partitioner class " + partitionerName, e);
```

Comment:
The change allows to
1. Set the correct `partitionerName` in the exception message. Previously, the message just include the partitioner set in the conf and it could be wrong when system property is present. 
2. Add the stack trace to the exception and enable logging.